scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2014 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+# author:  Imron Alston <imron@scalyr.com>
+
+__author__ = 'imron@scalyr.com'
+
+import datetime
+import os
+import re
+import select
+from scalyr_agent import ScalyrMonitor, define_config_option
+from scalyr_agent.scalyr_monitor import BadMonitorConfiguration
+from scalyr_agent.monitor_utils import journald_checkpoints
+from scalyr_agent.monitor_utils.journald_checkpoints import Checkpoint
+import scalyr_agent.scalyr_logging as scalyr_logging
+
+try:
+    from systemd import journal
+except ImportError:
+    raise Exception('Python systemd library not installed.\n\nYou must install the systemd python library in order '
+                    'to use the journald monitor.\n\nThis can be done via package manager e.g.:\n\n'
+                    '  apt-get install python-systemd  (debian/ubuntu)\n'
+                    '  dnf install python-systemd  (CentOS/rhel/Fedora)\n\n'
+                    'or installed from source using pip e.g.\n\n'
+                    '  pip install systemd-python\n\n'
+                    'See here for more info: https://github.com/systemd/python-systemd/\n'
+                   )
+
+global_log = scalyr_logging.getLogger(__name__)
+__monitor__ = __name__
+
+define_config_option( __monitor__, 'journal_path',
+                     'Optional (defaults to /var/log/journal). Location on disk of the journald logs.',
+                     convert_to=str, default='/var/log/journal')
+
+define_config_option( __monitor__, 'journal_poll_timeout_milliseconds',
+                     'Optional (defaults to 0). The number of milliseconds to wait for data while polling the journal file. '
```

Comment:
I think you should have a little explanation in the main documentation for the monitor on how the polling from journald works and reference it here.